constantine/ecdsa/ecdsa.nim

```diff
@@ -0,0 +1,356 @@
+import
+  ../hashes/h_sha256,
+  ../named/algebras,
+  ../math/io/[io_bigints, io_fields, io_ec],
+  ../math/elliptic/[ec_shortweierstrass_affine, ec_shortweierstrass_jacobian, ec_scalar_mul],
+  ../math/arithmetic,
+  ../platforms/abstractions,
+  ../serialization/codecs, # for fromHex and (in the future) base64 encoding
+  ../mac/mac_hmac, # for deterministic nonce generation via RFC 6979
+  ../named/zoo_generators, # for generator
+  ../csprngs/sysrand
+
+import std / macros # for `update` convenience helper
+
+type
+  ## Decides the type of sampler we use for the nonce. By default
+  ## a simple uniform random sampler. Alternatively a deterministic
+  ## sampler based on message hash and private key.
+  NonceSampler* = enum
+    nsRandom, ## pure uniform random sampling
+    nsRfc6979 ## deterministic according to RFC 6979
+
+# For easier readibility, define the curve and generator
+# as globals in this file
+const C* = Secp256k1
+const G = Secp256k1.getGenerator("G1")
```

Comment:
Yup, fair enough. I wrote it by defining the curve like this so that the code itself is already an (almost) valid generic / static proc. Didn't know if we want to make it fully generic immediately or start with secp256k1 only. Will address it.